### DISCUSSION TITLE
Why does the same JPEG image yield different dimensions when loaded with stbi_image on Mac and Android platforms?

### DISCUSSION THREAD

**Discussion by Adonis-Song** [nothings/stb#1599] [Q&A]
![cat1](https://github.com/nothings/stb/assets/34226411/3d106e4f-a9a3-44c0-b9a3-74749c63cca3)

Using this picture, on Mac, I get width = 4032, height = 3024. however on Android,  I get width =3024, height = 4032.

When I using ffmpeg to probe dimensions, I get  width =3024, height = 4032 . And the EXIF of this image does not indicate that it is a rotated picture.

**Comment by Adonis-Song** [comment]
[cat1.jpeg.zip](https://github.com/nothings/stb/files/14036673/cat1.jpeg.zip)
this is the zip of this picture

**Reply by nothings** [reply]
I don't know, most likely you're using it wrong on the Mac or actually loading a different image on Mac. stb_image has no code checks for Mac and contains no code to flip the image. It works correctly in my testbed (Windows), giving width=3024.
